### ISSUE TITLE
Issue: Clarification on the Usage of `last_index` for Preprocessing Chapman Data

### ISSUE THREAD

**Issue by yiminzme** [eth-siplab/Finding_Order_in_Chaos#4]
### Issue: Clarification on the Usage of `last_index` for Preprocessing Chapman Data

**Description:**

In the provided code, there is a segment where the `chapman_data` is being processed and subsequently sliced using `last_index`. This index is determined by finding the position of the first occurrence of the label `0` in `chapman_y`:
```python
last_index = np.argmax(chapman_y == 0)
chapman_x, chapman_y = chapman_x[0:last_index,:,:], chapman_y[0:last_index]
```
Given that the `chapman` dataset has only 4 label classes, the value of `last_index` is likely to be very small. As a result, the slicing operation may result in a very small subset of the `chapman_data`, which could potentially impact the training and evaluation process.

**Questions:**
1. Why is it necessary to slice the chapman_data at the first occurrence of label 0?
2. Can we directly split the entire chapman_x and chapman_y using train_test_split without slicing?

**Context:**
Here is the complete segment of the code for reference:
https://github.com/eth-siplab/Finding_Order_in_Chaos/blob/0d4b121322501efa101ce730198bfc996b1ddbe9/raw_data_process/CPSC%26Chapman/data_prep_combine.py#L28-L35

Your insights on this matter would be greatly appreciated.

**Additional Information:**
- The dataset being used is `chapman.mat`, and the labels are accessed via `chapman_labels`.
- The processing includes handling NaN values and preparing data for train-test split.
- There is a combination of two different datasets (`chapman` and `cpsc`) for experiments.

Thank you for your assistance!

---

**Comment by Berken-demirel** [comment]
Hi @yiminzme,

The indexing in the code is to check for any NaN values in the dataset, not to slice according to labels. As you can see from the code there are explicit NaN checkers for cpsc but not for chapman. The code was for debugging for any NaN values and you can just comment them out (it should work if there is no NaN values, I already edited with a comment). You can also see that this slicing is not for labels but for checking NaN values because in the line 44-45 from the script, the labels are normalized to one hot (0-1-2-3) from (1-2-3-4). In other words, 0 was assigned for NaN (if there is any).

https://github.com/eth-siplab/Finding_Order_in_Chaos/blob/0d4b121322501efa101ce730198bfc996b1ddbe9/raw_data_process/CPSC%26Chapman/data_prep_combine.py#L44-L45

You can access the Chapman dataset from [PhysioNet](https://physionet.org/content/ecg-arrhythmia/1.0.0/) and process with our provided [script](https://github.com/eth-siplab/Finding_Order_in_Chaos/blob/main/raw_data_process/CPSC%26Chapman/Chapman/chapman.m) and original scripts from the [paper](https://digitalcommons.chapman.edu/scs_articles/653/). 


To answer your questions after this clarification,
After ensuring that recordings from the same patient are not included in both the training and testing sets (preventing data leakage), you can split the data (as demonstrated in line 30). This approach is commonly used for ECG evaluation in contrastive learning, and you can find this in previous [works](https://arxiv.org/abs/2005.13249).

**Comment by yiminzme** [comment]
Thank you for the clarification. I now understand that the indexing in the code checks for NaN values rather than slicing by labels. I will comment out the NaN checkers as suggested, and use the provided scripts to process the Chapman dataset from PhysioNet.

Thanks again for your detailed response!
